e2e/tests/functional/plugins/styling/conditionSetStyling.e2e.spec.js

```diff
@@ -0,0 +1,170 @@
+
+    await searchAndLinkParameterToObject(page, stateGenerator.name, displayLayout.name);
+
+    //Add a box to the display layout
+    await page.goto(displayLayout.url, { waitUntil: 'domcontentloaded' });
+    await page.getByLabel('Edit Object').click();
+
+    //Add a box to the display layout and move it to the right
+    //TEMP: Click the layout such that the state generator is deselected
+    await page.getByLabel('Test Display Layout Layout Grid').locator('div').nth(1).click();
+    await page.getByLabel('Add Drawing Object').click();
+    await page.getByText('Box').click();
+    await page.getByLabel('X:').click();
+    await page.getByLabel('X:').fill('10');
+    await page.getByLabel('X:').press('Enter');
+
+    // set up conditional styling such that the box is red when the state generator condition is 'OFF'
+    await page.getByRole('tab', { name: 'Styles' }).click();
+    await page.getByRole('button', { name: 'Use Conditional Styling...' }).click();
+    await page.getByLabel('Modal Overlay').getByLabel('Expand My Items folder').click();
+    await page.getByLabel('Modal Overlay').getByLabel(`Preview ${conditionSet.name}`).click();
+    await page.getByText('Ok').click();
+    await page.getByLabel('Set background color').first().click();
+    await page.getByLabel('#ff0000').click();
+    await page.getByLabel('Save', { exact: true }).click();
+    await page.getByRole('listitem', { name: 'Save and Finish Editing' }).click();
+
+    await setRealTimeMode(page);
+
+    //Pause at a time when the state generator is 'OFF'
+    await page.clock.pauseAt(new Date(1732414800000));
+
+    const redBG = 'background-color: rgb(255, 0, 0);';
+    const defaultBG = 'background-color: rgb(102, 102, 102);';
+    const textElement = page.getByLabel('Alpha-numeric telemetry value').locator('div:first-child');
+    const styledElement = page.getByLabel('Box', { exact: true });
+
+    await expect(textElement).toHaveText('OFF');
+    await waitForStyleChange(styledElement, redBG);
+
+    // Fast forward to the next state change
+    await page.clock.fastForward(STATE_CHANGE_INTERVAL * 1000);
+
+    await expect(textElement).toHaveText('ON', { timeout: 2000 });
+
+    // Check if the style is not red when text is 'ON'
+    await waitForStyleChange(styledElement, defaultBG);
+  });
+});
+
+async function waitForStyleChange(element, expectedStyle, timeout = 0) {
+  await expect(async () => {
+    const style = await element.getAttribute('style');
+
+    // eslint-disable-next-line playwright/prefer-web-first-assertions
+    expect(style).toBe(expectedStyle);
+  }).toPass({ timeout }); // timeout allows for the style to be applied
+}
+
+/**
+ * Search for telemetry and link it to an object. objectName should come from the domainObject.name function.
+ * @param {import('@playwright/test').Page} page
+ * @param {string} parameterName
+ * @param {string} objectName
+ */
+async function searchAndLinkParameterToObject(page, parameterName, objectName) {
```

Comment:
this is a pre-promotion appAction. We're leveraging this in yamcsAppActions. I think it would be useful in core